### PR TITLE
test(e2e): modify addon e2e to test `storage init` with RDS

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR:
1. adds RDS option to `storage init` and enables flags needed for RDS
2. modify e2e addon test

Preceding PR: #2129 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: e2e/internal/client/cli.go [hunk 1]
```diff
@@ -407,6 +417,24 @@ func (cli *CLI) SvcLogs(opts *SvcLogsRequest) ([]SvcLogsOutput, error) {
...
+	--name $n
+	--storage-type $t
+	--workload $w
+	--engine $e
+    --initial-db $d
+*/
+func (cli *CLI) StorageInit(opts *StorageInitRequest)(string, error) {
+	return cli.exec(
+		exec.Command(cli.path, "storage", "init",
+			"--name", opts.StorageName,
```
Comment (iamhopaul123) [inline e2e/internal/client/cli.go:431]: nit: the format looks strange. it's not aligned

**Review by iamhopaul123 — COMMENTED** [review]
Instead of modifying the current addons e2e test suite, we might want to have a separate storage suite so as to include more storage types as well. I feel addons suite is more about simulating the user workflow when they create random addon files and use them. That why we have https://github.com/aws/copilot-cli/pull/2147/files#diff-bdb216deae877cb947e9f5ea259217b8ff55fd10b921cecd31ec4032b36f3356L108-L123. What do you think? (Plus next time we should separate flag changes in a separate PR)

File: internal/pkg/cli/storage_init.go [hunk 2]
```diff
@@ -755,6 +756,10 @@ are injected into your containers as environment variables for easy access.`,
 	cmd.Flags().BoolVar(&vars.noLSI, storageNoLSIFlag, false, storageNoLSIFlagDescription)
 	cmd.Flags().BoolVar(&vars.noSort, storageNoSortFlag, false, storageNoSortFlagDescription)
 
+	cmd.Flags().StringVar(&vars.rdsEngine, storageRDSEngineFlag, "", storageRDSEngineFlagDescription)
+	cmd.Flags().StringVar(&vars.rdsInitialDBName, storageRDSInitialDBFlag, "", storageRDSInitialDBFlagDescription)
+	cmd.Flags().StringVar(&vars.rdsParameterGroup, storageRDSParameterGroupFlag, "", storageRDSParameterGroupFlagDescription)
+
```
Comment (efekarakus) [inline internal/pkg/cli/storage_init.go:782]: Will these appear under `Aurora Serverless Flags` by any chance in the help menu? kinda like how we do for `DynamoDB Flags` that would be awesome

**Review by efekarakus — COMMENTED** [review]
Will the database be deleted on `app delete`? or do we `Retain` the resources?  

If so what do you think about removing the forced `Retain` from our addons?

**Comment by efekarakus** [comment]
> Instead of modifying the current addons e2e test suite, we might want to have a separate storage suite so as to include more storage types as well. I feel addons suite is more about simulating the user workflow when they create random addon files and use them. That why we have https://github.com/aws/copilot-cli/pull/2147/files#diff-bdb216deae877cb947e9f5ea259217b8ff55fd10b921cecd31ec4032b36f3356L108-L123. What do you think? (Plus next time we should separate flag changes in a separate PR)

I really like this change! The goal of our e2e tests like you said is to simulate users and we should aim to cover the most common usecases that we don't want to break. I think RDS and S3 with `storage init` would fit into this category for us. If we can add S3 in a separate PR, then we should cover our two most popular integrations! 

+ It will have the nice bonus of testing for breath with ec2 sec groups and iam managed roles from addon outputs.

**Comment by Lou1415926** [comment]
> Will the database be deleted on `app delete`? or do we `Retain` the resources?
> 
> If so what do you think about removing the forced `Retain` from our addons?

Talked offline but posting here for the records. The database will be deleted along with the add-on stack when the workload is deleted (e.g. by running `svc delete`). We didn't include `DeletionPolicy:Retain` in the template.

**Comment by Lou1415926** [comment]
> I really like this change! The goal of our e2e tests like you said is to simulate users and we should aim to cover the most common usecases that we don't want to break. I think RDS and S3 with `storage init` would fit into this category for us. If we can add S3 in a separate PR, then we should cover our two most popular integrations!
> 
>     * It will have the nice bonus of testing for breath with ec2 sec groups and iam managed roles from addon outputs.

A separate PR shall be created for S3 e2e testsuit! 

> Instead of modifying the current addons e2e test suite, we might want to have a separate storage suite so as to include more storage types as well. I feel addons suite is more about simulating the user workflow when they create random addon files and use them. That why we have https://github.com/aws/copilot-cli/pull/2147/files#diff-bdb216deae877cb947e9f5ea259217b8ff55fd10b921cecd31ec4032b36f3356L108-L123. What do you think? (Plus next time we should separate flag changes in a separate PR)

- I think having `storage init` (as opposed to copying files) is sort of like a super set of the user workflow when they bring their own add0on template. Can I think of it this way?
- Will separate flag changes next time

**Review by iamhopaul123 — APPROVED** [review]

**Review by efekarakus — APPROVED** [review]

**Pull request merged**